charts/openperouter/values.yaml

```diff
@@ -13,6 +13,10 @@ openperouter:
   # -- Controller log level. Must be one of: `debug`, `info`, `warn` or `error`.
   logLevel: info
   tolerateMaster: true
+  # -- If true, router and controller pods will run on master/control-plane nodes
+  routerOnMaster: true
+  # -- If true, nodemarker pod will run on master/control-plane nodes
+  nodeMarkerOnMaster: true
```

Comment:
Why should we have two different knobs ? While I understand we have multiple components in the system, IIUC, the controller / router components are daemonsets, thus scheduled in all workers nodes in the system. One of these knobs will allow these to be deployed in control-plane nodes.

The node labeler component, on the other hand, is a deployment which annotates the nodes. It is a deployment, running in a configurable subset of nodes. 

Thus, I think the **only** "knob" we actually require is a knob allowing single cluster installation, which would add the toleration to the control plane pods for both labeler, and controller.

Reply:
my reasoning was:

- a user might want or not to enhance his master nodes with what this router provides. For this, he might need to run the router / controller on masters too. For some scenarios, running on worker might be enough
- the nodemarker is a utility tool, and there's no value in running it on masters UNLESS you run a single node cluster

With one single knob, a user who wants to run the router on masters would have to take also the nodemarker running on masters even if he doesn't care about that.
If you think this is too much and the configuration becomes too complex, I can step back.

Reply:
I think it all boils down to if we envision people running the controller on control plane pods (what actually does the heavy lifting) but not wanting under any circumstance to have the nodemarker on the control plane. 

I don't see that happening, but maybe I'm short sighted. I would go with a single knob, `schedulableControlPlaneNodes` or something similar. 

Waiting for more people feedback. Having the two knobs is not a deal breaker for me.

Reply:
makes sense, we have room for making this more complex later. I'll switch to one single knob